Aarh first deliverable

AlanAndresRH opened the pull request.

AlanAndresRH committed:
First delivery Home module implementation

AlanAndresRH committed:
Second delivery Cryptocurrency app

AlanAndresRH committed:
Third delivery Cryptocurrency app

AlanAndresRH committed:
Third delivery Cryptocurrency app (configuration navController)

AlanAndresRH committed:
Completion of Room implementation in the entire project and implementation of navigation controller

AlanAndresRH committed:
Completion of Room implementation in the entire project and implementation of navigation controller

AlanAndresRH committed:
Completion of Room implementation in the entire project and implementation of navigation controller

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/data/repositories/BookRepositoryImpl.kt:
```diff
@@ -0,0 +1,79 @@
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class BookRepositoryImpl @Inject constructor(
```
Ten cuidado con los nombres. Si usas la notación ___Impl, la primera parte debe ser igual a la interfaz. Es decir, deben conservar el mismo prefijo, y quedarían CryptocurrencyRepository / CryptoCurrencyRepositoryImpl o BookRepository / BookRepositoryImpl pero no una combinación de ambas.

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/data/source/local/entities/AsksBookEntity.kt:
```diff
@@ -0,0 +1,24 @@
+    val price: String? = null
+)
+
+fun AsksBookEntity.toAskBidBook(): AskBid {
```
Este método podría llamarse solamente toAskBid()

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/data/source/local/entities/BidsBookEntity.kt:
```diff
@@ -0,0 +1,24 @@
+    val price: String? = null
+)
+
+fun BidsBookEntity.toAskBidBook(): AskBid {
```
Este método también podría llamarse solamente toAskBid()

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/data/source/local/dao/BookDao.kt:
```diff
@@ -0,0 +1,17 @@
+interface BookDao {
+
+    @Query("SELECT * FROM bookEntity")
+    suspend fun getCryptocurrencies(): List<BookEntity>
```
Como buena práctica, busca que el nombre del método tenga correspondencia con el objeto que regresa. Por ejemplo, para este caso es más conveniente nombrar los métodos getBooks(), insertBooks, ....

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/data/source/remote/dto/Payload.kt:
```diff
@@ -0,0 +1,13 @@
+package com.example.finalprojectwizelinecryptocurrencies.data.source.remote.dto
+
+data class Payload(
```
Por convención, agrega el sufijo a Dto a todos los modelos en data/remote. Esto te permite identificarlos fácilmente. Esta clase puede llamarse PayloadDto.

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/data/source/CryptocurrencyApi.kt:
```diff
@@ -0,0 +1,23 @@
+package com.example.finalprojectwizelinecryptocurrencies.data.source
```
Esta clase es específica de remote. Por favor, colócala dentro de la carpeta remote.

AlanAndresRH replied:
Disculpa no alcanzo a ver de que clase me habla

AlanAndresRH replied:
Listo, ya vi encontre cual es la clase a la que se refería. se corrige la nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/ui/detail/viewModel/DetailViewModel.kt:
```diff
@@ -0,0 +1,95 @@
+                state.update { state ->
+                    state.copy(
+                        book = resSuccess,
+                        isLoading = true
```
En este punto, el resultado ya fue cargado, por lo tanto, isLoading debería ser false

AlanAndresRH replied:
Este lo deje para que se vea la el progress Bar hasta que cargara los datos del servicio de Bids y Asks en el Recyclerview. Pero lo puedo quitar, como usted me diga.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/ui/detail/viewModel/DetailViewModel.kt:
```diff
@@ -0,0 +1,95 @@
+    val state = MutableStateFlow(DetailState(isLoading = true))
+
+    // TODO: Preguntar porque se muestra otra imagen primero y luego ya la verdadera imagen (Solo muestra la verdara imagen si comento estas lineas de código)
+    /*init {
```
En este punto no has asignado aún un valor a state.value.book.book, por lo tanto siempre llamarás getDetail("") y tendrás una imagen de error en la ui. Para que puedas cargar los valores desde el viewModel, deberás asignar primero el valor de book con los args en la navegación.

AlanAndresRH replied:
Ok, entonces. ¿ Lo puedo dejar como esta actualmente ?

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/ui/detail/DetailCryptocurrencyFragment.kt:
```diff
@@ -0,0 +1,87 @@
+            findNavController().popBackStack()
+        }
+        binding.rvAsks.apply {
+            hasFixedSize()
```
Utiliza por favor el método setHasFixedSize(true). El método hasFixedSize() es un getter y solo te devuelve un valor.

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo commented on app/src/main/java/com/example/finalprojectwizelinecryptocurrencies/ui/detail/adapters/AsksBidsAdapter.kt:
```diff
@@ -0,0 +1,43 @@
+        }
+
+        override fun areContentsTheSame(oldItem: AskBid, newItem: AskBid): Boolean {
+            return oldItem.book == newItem.book
```
La comparación que realizas aquí es sobre todo el elemento. Si, por ejemplo, en la lista cambia el precio sin que cambie el book, esta condición de areContentsTheSame seguirá regresando true, no detectará el cambio y no actualizará la lista. Por lo tanto, la condición aquí es generalmente oldItem == newItem.
(*Aplica también en CryptocurrencyAdapter.kt)

AlanAndresRH replied:
Listo, se corrige la siguiente nota.

diego-parra-robayo reviewed (commented):
Gran trabajo! Siento que vas avanzando muy bien en este proyecto. Te hice algunas notas en algunas líneas, y por favor, ten cuidado con los nombres, puede parecer trivial, pero los nombres adecuados te ayudan a identificar partes de tu código mucho más rápido a ti y a tu equipo.

AlanAndresRH committed:
Correction of observations in review of the first pull request

AlanAndresRH committed:
Correction of observations in review of the first pull request 2
HttpLoggingInterceptor is added and also the User-Agent header
Logic fix in AsksBookEntity and BidsBookEntity entity

AlanAndresRH commented:
AARH_Second_Deliverable

AlanAndresRH commented:
AARH_Second_Deliverable

AlanAndresRH committed:
Debugging warnings and errors
